pandas/io/excel/_base.py

```diff
@@ -439,7 +482,16 @@ def parse(
             else:  # assume an integer if not a string
                 sheet = self.get_sheet_by_index(asheetname)
 
-            data = self.get_sheet_data(sheet, convert_float)
+            gsd_header = 0 if header is None else header
```

Comment:
what does gsd mean? please make meaningful variable names

Reply:
it's supposed to be short for get_sheet_data, but I guess it's not clear